### ISSUE TITLE
App icon not shown on Wayland

### ISSUE THREAD

**Issue by MahshidX** [retorquere/zotero-deb#106]
App icon not shown when using the app on Wayland on Ubuntu **Dock**. (The issue is resolved if you switch to Xorg)

![image](https://github.com/user-attachments/assets/12b83df4-7df7-4454-9f66-e4732882c885)

Version: `7.0.0-beta.113+151218e4d (64-bit)`

**Comment by retorquere** [comment]
I'll have to see whether I can get Wayland up. I have an NVidia card and it's a PITA to get Wayland to work.

**Comment by Xashyar** [comment]
If you have 24.04, it automatically switches to Wayland, if you set `prime-select` to 'Intel' for instance.

**Comment by retorquere** [comment]
I don't know what that means.

**Comment by Xashyar** [comment]
Simply running the command `sudo prime-select intel` should do the job. Perhaps a simple reboot would be needed afterwards.

<details>
<summary><b>Details</b></summary>

Recent versions of Ubuntu default to Wayland on non-Nvidia hardware,` prime-select` is a tool for using [Optimus](https://en.wikipedia.org/wiki/Nvidia_Optimus) technology, which allows you to switch away from Nvidia graphics.

</details>

**Comment by retorquere** [comment]
I get

```
Error: no integrated GPU detected.
```

I'll see what I can do when I get back to the system, I may have disabled the iGPU. I'm considering just getting a new system. The iGPU is not supported by Win11, and getting a system where the iGPU is supprted by Win11 and by Wayland would just make things easier.

**Comment by epagone** [comment]
I have the same problem with Linux Mint Victoria Cinnamon edition with Xorg 😕

**Comment by dawaltconley** [comment]
After upgrading to 7 on Pop_OS, I had to change this line in `/opt/zotero/zotero.desktop` to get the icon to show up:

```desktop
# Icon=zotero.ico
Icon=/opt/zotero/icons/icon128.png
```

**Comment by retorquere** [comment]
> Not sure if related: after upgrading to 7 on Pop_OS (running X11, not wayland), I had to change this line in `/opt/zotero/zotero.desktop` to get the icon to show up:
> 
> ```desktop
> # Icon=zotero.ico
> Icon=/opt/zotero/icons/icon128.png
> ```

The icon path was indeed wrong in the zotero 7 package; 7.0-1 is out that fixes this.

**Comment by retorquere** [comment]
(that won't fix the problem @Xashyar is experiencing though, just the newly released Zotero 7)

**Comment by epagone** [comment]
> > Not sure if related: after upgrading to 7 on Pop_OS (running X11, not wayland), I had to change this line in `/opt/zotero/zotero.desktop` to get the icon to show up:
> > ```desktop
> > # Icon=zotero.ico
> > Icon=/opt/zotero/icons/icon128.png
> > ```
> 
> The icon path was indeed wrong in the zotero 7 package; 7.0-1 is out that fixes this.

Thanks, I confirm that in 7.0-1 my problem is fixed. Sorry for using this issue: it appeared to report the same problem that I was experiencing.

**Comment by retorquere** [comment]
NP. @Xashyar I still haven't been able to set up wayland. I'm looking for a radeon card to install so that I don't have to deal with nvidia.

**Comment by Xashyar** [comment]
@retorquere This shouldn't be that hard. If you log out from Ubuntu, In the bottom-right corner it should give you an option to log-in via Wayland or Xorg. See:

<img src="https://github.com/user-attachments/assets/30a4d14d-0c62-4b6f-9c3b-49245b78c19f" width="400" height="200">

**Comment by retorquere** [comment]
> I get
> 
> ```
> Error: no integrated GPU detected.
> ```
> 
> I'll see what I can do when I get back to the system, I may have disabled the iGPU. I'm considering just getting a new system. The iGPU is not supported by Win11, and getting a system where the iGPU is supprted by Win11 and by Wayland would just make things easier.

The iGPU is enabled and configured to allow fallback in the BIOS (or whatever it's called these days), but I get this error, and that's probably why I don't have that option at the bottom right.

lshw nor lspci show the iGPU. I'm looking at options for a radeon card or a SFF system with AMD graphics, I don't really want to spend time fighting with drivers.

**Comment by paulasimoes** [comment]
Not sure if it's the same problem, but I had no icon (like @Xashyar) both in Wayland and Xorg on Ubuntu dock (also I don't have an icon on "Show apps", although a search finds it). I went to the installation folder (in my case /lib/zotero) and run set_launcher_icon and now I have the icon on the dock (still not on show apps).

Version: 7.0.3 (64-bit) 
OS: Ubuntu 24.04 LTS

(@retorquere thanks for this package)

**Comment by retorquere** [comment]
But that script modifies `/usr/lib/zotero/zotero.desktop`. What happens if you rename the file to zotero.desktop.old? Maybe it will then pick up the desktop file that the package provides.

**Comment by retorquere** [comment]
Can you update/upgrade (there is a `-1` package now) to see if that works? In that package I rewrite the bundled `zotero.desktop`.

**Comment by retorquere** [comment]
The potential fix has been out 4 days. I'd appreciate feedback.

**Comment by paulasimoes** [comment]
Sorry for the delay, I've updated/upgraded. I get the icon on the dock as before, still no icon on the "show apps". I'm not noticing any change.

**Comment by Xashyar** [comment]
Using `7.0.3-beta.3+6f9abaaff (64-bit)` didn't notice any change either.

**Comment by retorquere** [comment]
you should be using `7.0.3-beta.3+6f9abaaff-1`.

**Comment by retorquere** [comment]
If I run `sudo ./set_launcher_icon`, the `zotero.desktop` does not change in the `-1` packages

> I went to the installation folder (in my case /lib/zotero)

this is not where my packages install zotero. Maybe you are not using my packages.

**Comment by Xashyar** [comment]
> you should be using `7.0.3-beta.3+6f9abaaff-1`.

I still see the same problem on `7.0.4-beta.2+974ac7861 (64-bit)`, should I update to the `-1` version differently?

**Comment by retorquere** [comment]
No, update should be automatic, but 7.0.4-beta.2+974ac7861 supersedes 7.0.3-beta.3+6f9abaaff-1. 

Does the contents of /usr/lib/zotero-beta/zotero.desktop change when you run set_launcher_icon there (it shouldn't), and does running set_launcher_icon get the icon to display?

**Comment by paulasimoes** [comment]
> this is not where my packages install zotero. Maybe you are not using my packages.

I'm sorry if I made you waste time. I did install the package from Zotero website first, but never made it work. Then I saw your deb and installed it and it worked (save for the icon). I did apt remove, purge, and delete a directory I thought it was the previous installation (the official Zotero). But maybe I didn't remove it properly. I'm sorry.

**Comment by retorquere** [comment]
what does `locate zotero.desktop` and `locate zotero-beta.desktop` get you?

**Comment by paulasimoes** [comment]
the first gets:
/home/username/.local/share/applications/zotero.desktop
/usr/lib/zotero/zotero.desktop
/usr/share/applications/zotero.desktop
the second does not get anything

**Comment by retorquere** [comment]
And if you remove /home/username/.local/share/applications/zotero.desktop, log out, log back in? This one is created by nor by the package, nor by set_launcher_icon.

**Comment by retorquere** [comment]
the locate database is not updated instantly.

**Comment by retorquere** [comment]
I've finally got the box that wayland will run on. I've done a clean installation of Ubuntu 24.04, and in that I installed `zotero` and `zotero6`, and the icons both show up in search and in the dock.

**Comment by Xashyar** [comment]
Hi @retorquere, I'm still facing the issue running zotero latest, are u sure that Ubuntu's running wayland? 24.04 will default to xorg if your system has nvidia graphics.

**Comment by retorquere** [comment]
```
$ echo $XDG_SESSION_TYPE
wayland
```

**Comment by retorquere** [comment]
this system has AMD graphics, I replaced the former machine.

**Comment by retorquere** [comment]
double-checked, I could choose "Ubuntu" or "Ubuntu on Xorg", chose Ubuntu, icon shows up.

**Comment by retorquere** [comment]
Ah that explains it. This will require a [change](https://groups.google.com/g/zotero-dev/c/J5ysVDXRmrA) by the Zotero team, but they've not been active on that group lately, it might take a while. If I can't move them, I'll just set the icon to the zotero (non-beta) icon.

**Comment by retorquere** [comment]
This still requires Zotero changes. See the link in my last message. There is nothing I can do about this.

**Comment by Darkluna999** [comment]
> This still requires Zotero changes. See the link in my last message. There is nothing I can do about this.这仍然需要修改 Zoonline。请参阅我最后一条消息中的链接。我对此无能为力。

Ah yes, but when I use beta manual installed is right. Maybe you can modify the desktop file or the script?

**Comment by retorquere** [comment]
That's because the tarball zotero beta places a `zotero.desktop` file rather than a `zotero-beta.desktop`, which makes sense for their situation as they don't account for having both zotero release and zotero beta installed. I can't safely write the same file (`zotero.desktop`) from different packages (`zotero` and `zotero-beta`). If they make the change I requested this will be automatically fixed.

**Comment by Darkluna999** [comment]
> That's because the tarball zotero beta places a `zotero.desktop` file rather than a `zotero-beta.desktop`, which makes sense for their situation as they don't account for having both zotero release and zotero beta installed. I can't safely write the same file (`zotero.desktop`) from different packages (`zotero` and `zotero-beta`). If they make the change I requested this will be automatically fixed.

Okay, thanks a lot for your explanation

**Comment by retorquere** [comment]
Copying `/usr/share/applications/zotero-beta.desktop` to `/usr/share/applications/zotero.desktop` will do the job in the interim.

For clarification, I could technically make it so that only one of the two can be installed and they would both deliver just `zotero`, but I actively use side-by-side installations.
